src/server/game/Scripting/ScriptMgr.h

```diff
@@ -879,6 +880,24 @@ class TC_GAME_API SceneScript : public ScriptObject
         virtual void OnSceneComplete(Player* /*player*/, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) { }
 };
 
+class TC_GAME_API ConversationScript : public ScriptObject
+{
+    protected:
+    // Manages registration, loading, and execution of scripts.
+        ConversationScript(const char* name);
+    
+    public:
+    
+        // Called when Conversation is created but not added to Map yet
+        virtual bool OnConversationCreate(Conversation* /*conversation*/, Unit* /*creator*/) { return true; }
+
+        // Called when an Conversation::AddActor is used
+        virtual void OnAddActor(Conversation* /*conversation*/, ObjectGuid /*actorGuid*/, uint16 /*actorIdx*/) { }
+
+        // Called when a Participant is added to conversation
+        virtual void OnAddParticipant(Conversation* /*conversation*/, ObjectGuid /*participantGuid*/) { }
```

Comment:
can use provide an example where OnAddActor and OnAddParticipant is useful for?

Reply:
No idea yet. But thought it might be useful for someone